pkg/cloud/amazon/common.go

```diff
@@ -82,10 +126,22 @@ func GetCurrentlyConnectedRegionAndClusterName() (string, string, error) {
 	if err != nil {
 		return "", "", errors.Wrapf(err, "loading kubeconfig")
 	}
+
 	context := kube.Cluster(kubeConfig)
-	currentClusterName, currentRegion, err := ParseContext(context)
+	server := kube.CurrentServer(kubeConfig)
+	currentClusterName, err := GetClusterNameFromAWS("", "", server)
+	if err != nil {
+		currentClusterName, currentRegion, err := ParseContext(context)
+		if err != nil {
+			return "", "", errors.Wrapf(err, "parsing the current Kubernetes context %s", context)
+		}
+
+		return currentClusterName, currentRegion, nil
+	}
+
+	currentRegion, err := ResolveRegionWithoutOptions()
```

Comment:
The big problem with trying to obtain the region/clustername from the context name is the fact that the context name could possibly be anything. This is why this issue was opened in the first place. I don't think its possible in every case to assume that we can use the context to obtain this information.
I could reorder the logic to attempt to use the context name first rather than the aws-sdk, if you prefer, but i believe that the sdk logic should remain.

Reply:
The thing is that the region value in the ARN will always match the region value used with the aws sdk's config. I think it adds unneeded code complexity by trying to parse the arn just for the sake of the region.

Maybe what i could do is use the active aws sdk session used for `DescribeCluster` and get the region from that sessions config, rather than calling `ResolveRegionWithoutOptions` and having that instantiate a new sdk session, just for the sake of pulling the region from the session config.

Reply:
Keep in mind that the person who created the cluster in a certain region with a certain configuration might not be the same person that is going to run install / boot later.

We need to be idempotent here and always return the region where the cluster was created and not the region that the user calling the command has configured.

Reply:
I understand, But if you are running `DescribeClusters` within aws region us-east-1 for an example, you will not be able to see any clusters that are in different aws regions. 
Therefore with that logic, the region the user has configured, must be the region that the cluster exists in. 
I think to be idempotent as you are suggesting, we need to have a command line flag or configuration value to allow reading the region from there that can override the users configuration value.

Reply:
You are right, that should be fine. I overlooked that part.